### PR TITLE
Add new saves localization for color-lines

### PR DESCRIPTION
**Pull request by Ra-V2**
Files saves now in /boot/home/color_lines/

### CODE REVIEW

File: games-puzzle/color-lines/patches/color_lines-0.6.patchset [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
++#include <sys/stat.h>
+ #endif
+ 
+ #ifndef WINDOWS
+@@ -1636,12 +1638,17 @@ char *get_path(int n)
+ 	static char path[255];
+ 
+ #ifndef WINDOWS
++#ifdef __HAIKU__
++mkdir("/boot/home/color_lines");
```
Comment (pulkomandy) [inline games-puzzle/color-lines/patches/color_lines-0.6.patchset]: use find_directory to locate the B_USER_SETTINGS_DIRECTORY, and store the data there.

File: games-puzzle/color-lines/patches/color_lines-0.6.patchset [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+diff --git a/main.c b/main.c
+index 4e79efe..bac137b 100644
+--- a/main.c
++++ b/main.c
+@@ -1639,7 +1639,7 @@ char *get_path(int n)
+ 
+ #ifndef WINDOWS
+ #ifdef __HAIKU__
+-mkdir("/boot/home/color_lines");
++mkdir("/boot/home/color_lines", 0700);
```
Comment (pulkomandy) [inline games-puzzle/color-lines/patches/color_lines-0.6.patchset]: This should be in B_USER_CONFIG_DIRECTORY as well.

File: games-puzzle/color-lines/patches/color_lines-0.6.patchset (hunk 1 above)
Comment (scottmc) [inline games-puzzle/color-lines/patches/color_lines-0.6.patchset]: avoid hard coding path names.  See https://github.com/haikuports/haikuports/wiki/FindDirectory

**Comment by Begasus** [comment]
Try to create a patchset that doesn't include previous patches (your patching patches in there atm)

1. clean your work "haikuporter -c *"
2. disable the patchset in the recipe
3. unpack the sources without building "haikuporter -b *"
4. patch the source, and commit, then move you current patchset to for instance the Desktop
5. extract the patchset "haikuporter -e *"
6. clean the build (1) enable the patchset in the recipe again (2) and then rebuild to check

File: games-puzzle/color-lines/patches/color_lines-0.6.patchset [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ 	if (gfx_init())
+ 		return 1;
+@@ -1630,12 +1639,21 @@ char *get_path(int n)
+ 	static char path[255];
+ 
+ #ifndef WINDOWS
++#ifdef __HAIKU__
++dev_t volume = dev_for_path("/boot");
++char buffer[B_PATH_NAME_LENGTH+B_FILE_NAME_LENGTH];
++find_directory(B_USER_SETTINGS_DIRECTORY, volume, true, buffer, sizeof(buffer));
```
Comment (korli) [inline games-puzzle/color-lines/patches/color_lines-0.6.patchset]: Please check for success.

File: games-puzzle/color-lines/patches/color_lines-0.6.patchset [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
++strcat(buffer, "/color_lines");
```
Comment (korli) [inline games-puzzle/color-lines/patches/color_lines-0.6.patchset]: please use strlcat

File: games-puzzle/color-lines/patches/color_lines-0.6.patchset [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
++mkdir(buffer, 0700);
++strcpy(path, buffer);
```
Comment (korli) [inline games-puzzle/color-lines/patches/color_lines-0.6.patchset]: please use strlcpy

File: games-puzzle/color-lines/patches/color_lines-0.6.patchset [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+ 		snprintf(path, sizeof(path) - 32, "%s", pw->pw_dir);
++#endif
+ #else
+ 	static char appdir[MAX_PATH];
+ 
+@@ -1657,13 +1675,13 @@ char *get_path(int n)
+ 
+ 	switch (n) {
+ 	case 0:
+-		strcat(path,"/.lines-save");
```
Comment (korli) [inline games-puzzle/color-lines/patches/color_lines-0.6.patchset]: please keep the upstream code with #ifndef __HAIKU__   ... #else ... #endif

**Comment by Begasus** [comment]
LGTM

**Pull request merged**